### ISSUE TITLE
[JIT] Python operations (e.g. Dropout) traced in wrong mode

### ISSUE THREAD

**Issue by jamesr66a** [pytorch/pytorch#6576]
Code example:

```
import torch
import torch.nn.functional as F

@torch.jit.trace(torch.zeros(1, 2, 3))
def dropout_test(x):
    x = torch.neg(x)
    return F.dropout(x)

dropout_test(torch.zeros(1, 2, 3, requires_grad=True))
```

Output:
```
17:20:04 ~ $ python dropoutcrash.py
Traceback (most recent call last):
  File "dropoutcrash.py", line 9, in <module>
    dropout_test(torch.zeros(1, 2, 3, requires_grad=True))
RuntimeError:
VariableType::ID() not implemented:
operation failed in interpreter:
/Users/jamesreed/onnx-fairseq/pytorch/torch/nn/functional.py(555): dropout
dropoutcrash.py(7): dropout_test
/Users/jamesreed/onnx-fairseq/pytorch/torch/jit/__init__.py(475): wrapper
dropoutcrash.py(4): <module>
```

Discussed with @zdevito offline, this appears to the the tracer tracing this op in the wrong mode, and two things are incorrect:

1) The `tracing_autograd_python_function` flag is set
2) The op has a Handle output

**Comment by deepbrain** [comment]
I was wondering if there was a version of the code where the dropout worked with the JIT? I reported similar issues with the dropout last December and was hoping that it would work one day. My current net is 99% CPU bound in python, especially at the inference, so the GPU is almost idle while the CPU is hogged at 1000%. The JIT really helped when it worked. Thanks!

**Comment by apaszke** [comment]
It used to work a few months ago, but apparently we didn't have this covered by our tests. BTW can you try rebuilding from master? It should hopefully be ok now.

Just out of curiosity, what's the perf difference you got from using the JIT?

**Comment by deepbrain** [comment]
I rebuilt from the latest main branch and am still getting the same as in my last test:
Function.apply returned the wrong number of outputs.:
operation failed in interpreter:
/home/tester/anaconda3/lib/python3.6/site-packages/torch/nn/functional.py(554): dropout
/home/tester/anaconda3/lib/python3.6/site-packages/torch/nn/modules/dropout.py(53): forward

when it was working, I was getting a very significant 2-3x improvement in inference and about 10% in training. Unfortunately it was not working correctly with multiple GPUs - the other issue I posted in December and today again, so the code was never useful for me. I was wondering if these simple tests could be added to your regression tests?
